File: easy/145_binary_tree_postorder_traversal.rb
# 145. Binary Tree Postorder Traversal
# Easy
#
# Given the root of a binary tree, return the postorder traversal of its
# nodes' values.

class TreeNode
  attr_accessor :val, :left, :right

  def initialize(val = 0, left = nil, right = nil)
    @val = val
    @left = left
    @right = right
  end
end

def postorder_traversal(root)
  return [] if root.nil?

  res = postorder_traversal(root.left)
  res.concat(postorder_traversal(root.right))
  res << root.val
  res
end

File: go/easy/145_binary_tree_postorder_traversal.go
// 145. Binary Tree Postorder Traversal
// Easy
//
// Given the root of a binary tree, return the postorder traversal of its
// nodes' values.
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func postorderTraversal(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}

	res := postorderTraversal(root.Left)
	res = append(res, postorderTraversal(root.Right)...)
	return append(res, root.Val)
}
